app/scripts/models/reliers/oauth.js

```diff
@@ -204,6 +207,51 @@ define(function (require, exports, module) {
     },
 
     /**
+     * Check if the relier wants access to the account encryption keys.
+     *
+     * @returns {Boolean}
+     */
+    wantsKeys () {
+      return !! (this._config && this._config.scopedKeysEnabled && this._validateKeyScopeRequest());
+    },
+
+    /**
+     * Validate the requested scope with the relier redirect uri.
+     * At least one valid match must be found to successfully validate
+     * @returns {boolean}
+     * @private
+     */
+    _validateKeyScopeRequest () {
+      if (! this.has('keysJwk')) {
+        return false;
+      }
+
+      const validation = this._config.scopedKeysValidation || {};
+      let foundRedirectScopeMatch = false;
+
+      if (! this.get('scope')) {
+        throw new Error('Invalid scope parameter');
+      }
+
+      scopeStrToArray(this.get('scope')).forEach((scope) => {
+        if (validation.hasOwnProperty(scope)) {
+          if (validation[scope].redirectUris.includes(this.get('redirectUri'))) {
+            foundRedirectScopeMatch = true;
+          } else {
+            throw new Error('Invalid redirect parameter');
+          }
+        }
+      });
+
+      if (! foundRedirectScopeMatch) {
+        throw new Error('Scope not supported');
```

Comment:
nit: the error message here doens't seem to reflect that actually went wrong, something like 'no key-bearing scopes were requested' may be clearer.